bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/RuleEnabledHandler.java

```diff
@@ -0,0 +1,160 @@
+     */
+    public static final String MODULETYPE_ID = "RuleEnableAction";
+
+    /**
+     * This field is a key to the 'action' property of the {@link Action}.
+     */
+    private final static String ACTION_KEY = "action";
+
+    /**
+     * This field is a key to the 'ruleIds' property of the {@link Action}.
+     */
+    private final static String RULE_UIDS_KEY = "ruleIds";
+
+    /**
+     * This field contains the filter to track the {@link RuleRegistry} service.
+     */
+    private final static String FILTER = "(objectClass=" + RuleRegistry.class.getName() + ")";
+
+    /**
+     * This field will store the IDs of the rules to which the action will be applied.
+     */
+    private String[] rulesIds;
+
+    /**
+     * This field will store the action that should be applied.
+     */
+    private boolean enabled;
+
+    /**
+     * This Tracker is used to manage ruleAdmin field.
+     */
+    private ServiceTracker tracker = null;
+
+    /**
+     * This field will be used to manage the rules.
+     */
+    private RuleRegistry registry = null;
+
+    /**
+     * Reference to the bundle context of the framework.
+     */
+    private BundleContext context = null;
+
+    /**
+     * This field will be true if at some point in execution of this {@link Action} the {@link RuleRegistry} becomes
+     * unavailable.
+     */
+    private boolean isInterrupted = false;
+
+    @SuppressWarnings({ "unchecked" })
+    public RuleEnabledHandler(Action module, BundleContext context) {
+        super(module);
+        final Configuration config = module.getConfiguration();
+        if (config == null) {
+            throw new IllegalArgumentException("'Configuration' can not be null.");
+        }
+
+        String action = (String) config.get(ACTION_KEY);
+        if (action == null) {
+            throw new IllegalArgumentException("'action' property can't be undefined.");
+        }
+
+        if (action.equalsIgnoreCase("enable")) {
+            enabled = true;
+        } else if (action.equalsIgnoreCase("disable")) {
+            enabled = false;
+        } else {
+            throw new IllegalArgumentException(
+                    "permitted values for the 'action' property are 'enable' and 'disable'.");
+        }
+
+        final String rules = (String) config.get(RULE_UIDS_KEY);
+
+        if (rules == null) {
+            throw new IllegalArgumentException("'ruleIds' property can't be undefined.");
+        }
+        rulesIds = rules.split(",");
+
+        this.context = context;
+
+        try {
+            Filter filter = FrameworkUtil.createFilter(FILTER);
+            tracker = new ServiceTracker(context, filter, this);
+        } catch (InvalidSyntaxException e) {
+            // Should never happen.
+        }
+    }
+
+    @Override
+    public Map<String, Object> execute(Map<String, ?> context) {
+
+        try {
+            for (String entity : rulesIds) {
+                registry.setEnabled(entity, enabled);
```

Comment:
I think it is better to check if the registry is null instead of try/catch of NPE. When it is null just write in log warnings of not executed enable/disable rules